[toolchain] building SIRIUS fails when python is Python 3

For Python 3-only system, building SIRIUS via the toolchain fails with:

```
[...]
CMake Error at /data/tiziano/cp2k/tools/toolchain/install/cmake-3.15.1/share/cmake-3.15/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find Python2 (missing: Python2_EXECUTABLE Interpreter)
Call Stack (most recent call first):
  /data/tiziano/cp2k/tools/toolchain/install/cmake-3.15.1/share/cmake-3.15/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  cmake/modules/FindPythonSupport.cmake:1170 (find_package_handle_standard_args)
  cmake/modules/FindPython2.cmake:181 (include)
  CMakeLists.txt:45 (find_package)
```

For example on openSUSE Leap 15.1.

--- a/tools/toolchain/scripts/install_sirius.sh
+++ b/tools/toolchain/scripts/install_sirius.sh
@@ -133,7 +133,6 @@ case "$with_sirius" in
                    -DCMAKE_CXX_FLAGS_RELWITHDEBINFO="${SIRIUS_DBG}" \
                    -DCMAKE_CXX_COMPILER=mpic++ \
                    -DCMAKE_C_COMPILER=mpicc \
-                   -DPYTHON2=ON \
                   ${COMPILATION_OPTIONS} .. > compile.log 2>&1
              make -j $NPROCS -C src >> compile.log 2>&1
 
@@ -156,7 +155,6 @@ case "$with_sirius" in
                       -DCMAKE_CXX_FLAGS_RELWITHDEBINFO="${SIRIUS_DBG}" \
                       -DUSE_CUDA=ON \
                       -DGPU_MODEL=P100 \
-                      -DPYTHON2=ON \
                       -DCMAKE_CXX_COMPILER=mpic++ \
                       -DCMAKE_C_COMPILER=mpicc ${COMPILATION_OPTIONS} .. >> compile.log 2>&1
                 make -j $NPROCS -C src >> compile.log 2>&1

